epsagon/trace.py

```diff
@@ -761,8 +768,47 @@ def remove_ignored_keys(self, input_dict):
                     if isinstance(value, dict):
                         self.remove_ignored_keys(value)
 
-    # pylint: disable=W0703
     def send_traces(self):
+        """
+        If trace size exceeds the maximum size, and split flag is on
+        then split the trace into multiple traces.
+        :return: None
+        """
+        if self.split_on_send and self.length > self._max_trace_size:
+            self._send_trace_split()
+        else:
+            self._send_traces()
+
+    def _send_trace_split(self):
+        """
+        Split trace into multiple traces and send them one after the other.
+        This is done by manipulating the trace object while keeping the same
+        runner.
+        :param trace: the trace to send.
+        """
+        # Get only events (without runner)
+        all_events = self.events.copy()
```

Comment:
deepcopy?

Reply:
I don't need to deep copy the events objects. only the list (so it won't be affected when looping )